### PR TITLE
Implement stdin/stdout/stderr redirection support using multiport virtio-console

### PR DESCRIPTION
**Pull request by mtjhrc**

### CODE REVIEW

File: src/devices/src/virtio/console/console_control.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
+use std::collections::VecDeque;
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+
+use libc::EFD_NONBLOCK;
```
Comment (slp) [inline src/devices/src/virtio/console/console_control.rs:5]: Please use `utils::eventfd::EFD_NONBLOCK` instead, which is a wrapper for macOS support.

File: src/devices/src/virtio/console/port_io.rs [hunk 2]
```diff
@@ -0,0 +1,123 @@
+use std::io::{self, ErrorKind};
+use std::os::fd::{AsRawFd, FromRawFd, OwnedFd, RawFd};
+
+use libc::{fcntl, F_GETFL, F_SETFL, O_NONBLOCK, STDIN_FILENO, STDOUT_FILENO};
```
Comment (slp) [inline src/devices/src/virtio/console/port_io.rs]: Ditto.

File: src/devices/src/virtio/console/port_io.rs [hunk 3]
```diff
@@ -1,11 +1,14 @@
 use std::io::{self, ErrorKind};
 use std::os::fd::{AsRawFd, FromRawFd, OwnedFd, RawFd};
 
-use libc::{fcntl, F_GETFL, F_SETFL, O_NONBLOCK, STDERR_FILENO, STDIN_FILENO, STDOUT_FILENO};
+use libc::{
+    fcntl, EFD_NONBLOCK, F_GETFL, F_SETFL, O_NONBLOCK, STDERR_FILENO, STDIN_FILENO, STDOUT_FILENO,
```
Comment (slp) [inline src/devices/src/virtio/console/port_io.rs:5]: Same here with EFD_NONBLOCK.

File: src/vmm/src/builder.rs [hunk 4]
```diff
@@ -28,6 +29,7 @@ use kbs_types::Tee;
 use crate::device_manager;
 #[cfg(feature = "tee")]
 use crate::resources::TeeConfig;
+use crate::signal_handler::register_sigint_handler;
```
Comment (slp) [inline src/vmm/src/builder.rs:32]: Please put the `register_sigint_handler` import and its uses behind a `#[cfg(target_os = "linux")]` build-time conditional.

**Review by slp — CHANGES_REQUESTED** [review]
Overall looks good, just suggested some minor changes for macOS compatbility.

File: src/vmm/src/builder.rs (hunk 4 above)
Comment (mtjhrc) [reply src/vmm/src/builder.rs:32]: Oh the whole signal_handler file is not compiled on macOS. 

Btw we can `cargo clippy --target aarch64-apple-darwin` to the pipeline, which checks agains simple issues like this (non existing imports), but this currently generates some warnings, so maybe it should could be in another PR.

Comment (slp) [reply src/vmm/src/builder.rs:33]: Yeah, let's get this one merged first. I'll prepare a new PR setting up the CI for aarch64-apple-darwin.

**Review by slp — APPROVED** [review]
Builds now fine on macOS/aarch64. Let's get it merged. Thanks!

**Pull request merged**